### PR TITLE
feat: optionally throttle writes when exporting is too slow

### LINKED ISSUE #19681: Zeebe can adjust write rate limit based on exporting backlog

### PR DESCRIPTION
**Pull request by lenaschoenburg**
## Description

When `zeebe.broker.flowControl.write.throttling.enabled=true`, the write rate is throttled, based on the observed exporting rate.

When the exporting backlog is approaching the configurable `acceptableBacklog`, the write rate limit is gradually decreased. When exporting rate continues to be slow, the write rate settles to match exporting rate exactly, maintaining a constant exporting backlog of `acceptableBacklog`.

When writes are throttled, a new annotation marks this:
exporting slowed down
exporting backlog stablilizing
backpressure increases

## Related issues

closes #19681

### CODE REVIEW

File: zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/management/FlowControlEndpointIT.java [hunk 1]
```diff
@@ -34,105 +30,115 @@ final class FlowControlEndpointIT {
...
+        }""");
+    final var flowControlConfiguration = actuator.getFlowControlConfiguration();
 
     // then
-    assertThat(flowControlConfiguration.get(1).toString())
-        .contains(
-            "\"requestLimiter\":{\"limit\":50,\"estimatedLimit\":50.0,\"rtt_noload\":0,\"maxLimit\":1000,\"smoothing\":1.0}",
-            "\"writeRateLimit\":{\"enabled\":true,\"limit\":999,\"rampUp\":0.0}");
-  }
+    final var requestLimiter = flowControlConfiguration.get(1).get("requestLimiter");
```
Comment (rodrigo-lourenco-lopes) [inline zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/management/FlowControlEndpointIT.java:59]: Thanks for the refactor of this 🙌

File: zeebe/logstreams/src/main/java/io/camunda/zeebe/logstreams/impl/flowcontrol/RateLimit.java [hunk 2]
```diff
@@ -25,4 +38,25 @@ public RateLimiter limiter() {
     }
     return RateLimiter.create(limit, rampUp);
   }
+
+  public record Throttling(
```
Comment (rodrigo-lourenco-lopes) [inline zeebe/logstreams/src/main/java/io/camunda/zeebe/logstreams/impl/flowcontrol/RateLimit.java:42]: ❓ should we somehow assert that the throttling min rate is not higher than the rate limit if these are enabled?

**Review by rodrigo-lourenco-lopes — APPROVED** [review]
Thanks for the addition! 🚀 🚀

File: zeebe/logstreams/src/main/java/io/camunda/zeebe/logstreams/impl/flowcontrol/RateLimit.java (hunk 2 above)
Comment (lenaschoenburg) [reply zeebe/logstreams/src/main/java/io/camunda/zeebe/logstreams/impl/flowcontrol/RateLimit.java:45]: Good point, I added it here: ae02649

**Pull request merged**